NumLib/DOF/LocalToGlobalIndexMap.cpp

```diff
@@ -272,6 +272,37 @@ LocalToGlobalIndexMap::getNumberOfElementDOF(std::size_t const mesh_item_id) con
     return ndof;
 }
 
+std::size_t
+LocalToGlobalIndexMap::getNumberOfElementComponents(std::size_t const mesh_item_id) const
+{
+    std::size_t n = 0;
+    for (unsigned c=0; c<_rows.cols(); ++c)
+    {
+        if (!_rows(mesh_item_id, c).empty())
+            n++;
+    }
```

Comment:
(This is in principle std::count_if but I didn't find a concise way to access the iterators.)
Question, how often is getNumberOfElementComponents is called? I'm asking because if it is often (like for every iteration of the non-linear solver for all elements O(#E*#AvgIterations*#timesteps) then it would be worth to look into optimizations.
One thing is coming into my mind (but I didn't benchmark it!) is to iterate over `_rows.row(mesh_item_id)` object.